File: Graph/997. Find the Town Judge/997.py
class Solution:
    def findJudge(self, n: int, trust: List[List[int]]) -> int:
        people = [0] * (n+1)
        for a, b in trust: people[a] -= 1; people[b] += 1
        return next((i for i in range(1,n+1) if people[i] >= n-1), -1)

File: Graph/DFS/997. Find the Town Judge/997.java
class Solution {
    public int findJudge(int n, int[][] trust) {
        int[] people = new int[n+1];
        for(int[] pair : trust){
            people[pair[0]]--;
            people[pair[1]]++;
        }
        for(int i=1; i<=n; i++){
            if(people[i] >= n-1) return i;
        }
        return -1;
    }
}